extensions/mailer/deployment/src/main/java/io/quarkus/mailer/deployment/MailerProcessor.java

```diff
@@ -68,4 +87,50 @@ MailerBuildItem build(BuildProducer<FeatureBuildItem> feature, MailConfigRecorde
 
         return new MailerBuildItem(client);
     }
+
+    @BuildStep
+    void validateMailTemplates(
+            List<TemplatePathBuildItem> templatePaths, ValidationPhaseBuildItem validationPhase,
+            BuildProducer<ValidationErrorBuildItem> validationErrors) {
+
+        Set<String> filePaths = new HashSet<String>();
+        for (TemplatePathBuildItem templatePath : templatePaths) {
+            String filePath = templatePath.getPath();
+            if (File.separatorChar != '/') {
+                filePath = filePath.replace(File.separatorChar, '/');
+            }
+            if (filePath.endsWith("html") || filePath.endsWith("htm") || filePath.endsWith("txt")) {
+                // For e-mails we only consider html and txt templates
+                filePaths.add(filePath);
+                int idx = filePath.lastIndexOf('.');
+                if (idx != -1) {
+                    // Also add version without suffix from the path
+                    // For example for "items.html" also add "items"
+                    filePaths.add(filePath.substring(0, idx));
+                }
+            }
+        }
+
+        for (InjectionPointInfo injectionPoint : validationPhase.getContext().get(BuildExtension.Key.INJECTION_POINTS)) {
+            if (injectionPoint.getRequiredType().name().equals(MAIL_TEMPLATE)) {
+                AnnotationInstance resourcePath = injectionPoint.getRequiredQualifier(QuteProcessor.RESOURCE_PATH);
+                String name;
+                if (resourcePath != null) {
+                    name = resourcePath.value().asString();
+                } else if (injectionPoint.hasDefaultedQualifier()) {
+                    name = QuteProcessor.getName(injectionPoint);
+                } else {
+                    name = null;
```

Comment:
This is a valid case?

Reply:
It is, in theory. We do not cover injection points with `MailTemplate` required type and qualifiers other than `@ResourcePath` or `@Default` (implicit).